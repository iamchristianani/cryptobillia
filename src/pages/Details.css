.coinBox {
  width: 100%;
}

.coinHeaderBox {
  width: 100%;
  padding: 20px 20px;
  background: #5789e6;
  position: relative;
  display: flex;
  align-items: center;
}

.coinHeaderBox img {
  height: 70px;
  margin-left: 20px;
}

.coinHeaderBoxDetails {
  width: 100%;
  margin-left: 100px;
}

.coinHeaderBoxDetails h2 {
  color: gold;
}

.coinHeaderBoxDetails h4 {
  color: lemonchiffon;
}

.coinHeaderBoxDetails p {
  margin-top: 10px;
}

.allDetailsInfo {
  color: #fff;
  width: 375px;
}

.coinDetailsHeader {
  padding: 10px 20px;
  background: #35548b;
  color: gold;
}

.coinDetailsInfo {
  background: #3e61a5;
  padding: 10px 20px;
}

.coinDetailsInfo ul {
  list-style: none;
  display: flex;
  margin: 0;
  padding: 10px 0;
  gap: 20px;
  justify-content: center;
  align-items: center;
}

.detailsIcons {
  font-size: 20px;
  cursor: pointer;
}

table {
  width: 100%;
}

td {
  padding: 0 0 10px 0;
}

tr td:first-child {
  color: lemonchiffon;
}
