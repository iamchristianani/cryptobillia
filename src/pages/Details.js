import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import './Details.css';
import { AiFillFacebook, AiFillGithub, AiFillTwitterCircle, AiFillRedditCircle } from 'react-icons/ai';
import { getDetailsAction } from '../redux/coindetails/coindetails';

const Details = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.details);
  const { coinId } = useParams();

  useEffect(() => {
    dispatch(getDetailsAction(coinId));
  }, [dispatch, coinId]);

  const facebookUsername = data.links ? data.links.facebook_username : null;
  const facebookURL = "https://facebook.com/".concat(facebookUsername);
  const twitterUsername = data.links ? data.links.twitter_screen_name : null;
  const twitterURL = "https://twitter.com/".concat(twitterUsername);

  return (
    <div className='coinBox'>
      <div className="coinHeaderBox">
        <img src={data.image ? data.image.large : null} alt={data.name} />
        <div className="coinHeaderBoxDetails">
          <h2>{data.name}</h2>
          <h4>({data.symbol})</h4>
          {
            data.market_data?.current_price
            ? <p>
              $
              {data.market_data.current_price.usd}
            </p> : null
          }
        </div>
      </div>
      <div className="allDetailsInfo">
        <div className="coinDetails">
          <div className="coinDetailsHeader">
            <h3>Coin Details</h3>
          </div>
          <div className="coinDetailsInfo">
            <table>
              <tbody>
                <tr>
                  <td>Name: </td>
                  <td>{data.name}</td>
                </tr>
                <tr>
                  <td>USD Price: </td>
                  <td>${data.market_data?.current_price?data.market_data.current_price.usd : null}</td>
                </tr>
                <tr>
                  <td>EUR Price: </td>
                  <td>&#8364;{data.market_data?.current_price?data.market_data.current_price.eur : null}</td>
                </tr>
                <tr>
                  <td>GBP Price: </td>
                  <td>&#163;{data.market_data?.current_price?data.market_data.current_price.gbp : null}</td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div className="coinDetails">
          <div className="coinDetailsHeader">
            <h3>More Info</h3>
          </div>
          <div className="coinDetailsInfo">
          <table>
              <tbody>
                <tr>
                  <td>Market Rank: </td>
                  <td>{data.market_cap_rank}</td>
                </tr>
                <tr>
                  <td>Percentage Change: </td>
                  <td>{data.market_data?data.market_data.price_change_percentage_24h: null}</td>
                </tr>
                <tr>
                  <td>Circulating Supply: </td>
                  <td>{data.market_data?data.market_data.circulating_supply : null}</td>
                </tr>
                <tr>
                  <td>Market Supply: </td>
                  <td>{data.market_data?data.market_data.total_supply : null}</td>
                </tr>
                <tr>
                  <td>Market Cap: </td>
                  <td>${data.market_data?.market_cap?data.market_data.market_cap.usd : null}</td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div className="coinDetails">
          <div className="coinDetailsHeader">
            <h3>Socials</h3>
          </div>
          <div className="coinDetailsInfo">
            <ul>
              <li>
                <a href={facebookURL} target="_blank" rel="noreferrer">
                  <AiFillFacebook className="detailsIcons" />
                </a>
              </li>
              <li>
                <a href={twitterURL} target="_blank" rel="noreferrer">
                  <AiFillTwitterCircle className="detailsIcons" />
                </a>
              </li>
              <li>
                <a href={data.links?.repos_url ? data.links.repos_url.github[0] : null} target="_blank" rel="noreferrer">
                  <AiFillGithub className="detailsIcons" />
                </a>
              </li>
              <li>
                <a href={data.links ? data.links.subreddit_url : null} target="_blank" rel="noreferrer">
                  <AiFillRedditCircle className="detailsIcons" />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Details;
