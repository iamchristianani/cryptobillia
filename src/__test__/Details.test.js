import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Details from '../pages/Details';

describe('Details Test', () => {
  test('to render each coin details: ', () => {
    const details = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );
    expect(details).toMatchSnapshot();
  });
});
