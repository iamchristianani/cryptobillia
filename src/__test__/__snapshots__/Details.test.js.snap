// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Test to render each coin details:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coinBox"
      >
        <div
          class="coinHeaderBox"
        >
          <img />
          <div
            class="coinHeaderBoxDetails"
          >
            <h2 />
            <h4>
              (
              )
            </h4>
          </div>
        </div>
        <div
          class="allDetailsInfo"
        >
          <div
            class="coinDetails"
          >
            <div
              class="coinDetailsHeader"
            >
              <h3>
                Coin Details
              </h3>
            </div>
            <div
              class="coinDetailsInfo"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      Name: 
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      USD Price: 
                    </td>
                    <td>
                      $
                    </td>
                  </tr>
                  <tr>
                    <td>
                      EUR Price: 
                    </td>
                    <td>
                      €
                    </td>
                  </tr>
                  <tr>
                    <td>
                      GBP Price: 
                    </td>
                    <td>
                      £
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="coinDetails"
          >
            <div
              class="coinDetailsHeader"
            >
              <h3>
                More Info
              </h3>
            </div>
            <div
              class="coinDetailsInfo"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      Market Rank: 
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Percentage Change: 
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Circulating Supply: 
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Market Supply: 
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Market Cap: 
                    </td>
                    <td>
                      $
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="coinDetails"
          >
            <div
              class="coinDetailsHeader"
            >
              <h3>
                Socials
              </h3>
            </div>
            <div
              class="coinDetailsInfo"
            >
              <ul>
                <li>
                  <a
                    href="https://facebook.com/null"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      class="detailsIcons"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                      />
                    </svg>
                  </a>
                </li>
                <li>
                  <a
                    href="https://twitter.com/null"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      class="detailsIcons"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 0 1-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 0 1-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 0 0 229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                      />
                    </svg>
                  </a>
                </li>
                <li>
                  <a
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      class="detailsIcons"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                      />
                    </svg>
                  </a>
                </li>
                <li>
                  <a
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      class="detailsIcons"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M584 548a36 36 0 1 0 72 0 36 36 0 1 0-72 0zm144-108a35.9 35.9 0 0 0-32.5 20.6c18.8 14.3 34.4 30.7 45.9 48.8A35.98 35.98 0 0 0 728 440zM512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm245 477.9c4.6 13.5 7 27.6 7 42.1 0 99.4-112.8 180-252 180s-252-80.6-252-180c0-14.5 2.4-28.6 7-42.1A72.01 72.01 0 0 1 296 404c27.1 0 50.6 14.9 62.9 37 36.2-19.8 80.2-32.8 128.1-36.1l58.4-131.1c4.3-9.8 15.2-14.8 25.5-11.8l91.6 26.5a54.03 54.03 0 0 1 101.6 25.6c0 29.8-24.2 54-54 54-23.5 0-43.5-15.1-50.9-36.1L577 308.3l-43 96.5c49.1 3 94.2 16.1 131.2 36.3 12.3-22.1 35.8-37 62.9-37 39.8 0 72 32.2 72 72-.1 29.3-17.8 54.6-43.1 65.8zm-171.3 83c-14.9 11.7-44.3 24.3-73.7 24.3s-58.9-12.6-73.7-24.3c-9.3-7.3-22.7-5.7-30 3.6-7.3 9.3-5.7 22.7 3.6 30 25.7 20.4 65 33.5 100.1 33.5 35.1 0 74.4-13.1 100.2-33.5 9.3-7.3 10.9-20.8 3.6-30a21.46 21.46 0 0 0-30.1-3.6zM296 440a35.98 35.98 0 0 0-13.4 69.4c11.5-18.1 27.1-34.5 45.9-48.8A35.9 35.9 0 0 0 296 440zm72 108a36 36 0 1 0 72 0 36 36 0 1 0-72 0z"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coinBox"
    >
      <div
        class="coinHeaderBox"
      >
        <img />
        <div
          class="coinHeaderBoxDetails"
        >
          <h2 />
          <h4>
            (
            )
          </h4>
        </div>
      </div>
      <div
        class="allDetailsInfo"
      >
        <div
          class="coinDetails"
        >
          <div
            class="coinDetailsHeader"
          >
            <h3>
              Coin Details
            </h3>
          </div>
          <div
            class="coinDetailsInfo"
          >
            <table>
              <tbody>
                <tr>
                  <td>
                    Name: 
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    USD Price: 
                  </td>
                  <td>
                    $
                  </td>
                </tr>
                <tr>
                  <td>
                    EUR Price: 
                  </td>
                  <td>
                    €
                  </td>
                </tr>
                <tr>
                  <td>
                    GBP Price: 
                  </td>
                  <td>
                    £
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="coinDetails"
        >
          <div
            class="coinDetailsHeader"
          >
            <h3>
              More Info
            </h3>
          </div>
          <div
            class="coinDetailsInfo"
          >
            <table>
              <tbody>
                <tr>
                  <td>
                    Market Rank: 
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    Percentage Change: 
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    Circulating Supply: 
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    Market Supply: 
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    Market Cap: 
                  </td>
                  <td>
                    $
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="coinDetails"
        >
          <div
            class="coinDetailsHeader"
          >
            <h3>
              Socials
            </h3>
          </div>
          <div
            class="coinDetailsInfo"
          >
            <ul>
              <li>
                <a
                  href="https://facebook.com/null"
                  rel="noreferrer"
                  target="_blank"
                >
                  <svg
                    class="detailsIcons"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                    />
                  </svg>
                </a>
              </li>
              <li>
                <a
                  href="https://twitter.com/null"
                  rel="noreferrer"
                  target="_blank"
                >
                  <svg
                    class="detailsIcons"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 0 1-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 0 1-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 0 0 229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                    />
                  </svg>
                </a>
              </li>
              <li>
                <a
                  rel="noreferrer"
                  target="_blank"
                >
                  <svg
                    class="detailsIcons"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                    />
                  </svg>
                </a>
              </li>
              <li>
                <a
                  rel="noreferrer"
                  target="_blank"
                >
                  <svg
                    class="detailsIcons"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M584 548a36 36 0 1 0 72 0 36 36 0 1 0-72 0zm144-108a35.9 35.9 0 0 0-32.5 20.6c18.8 14.3 34.4 30.7 45.9 48.8A35.98 35.98 0 0 0 728 440zM512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm245 477.9c4.6 13.5 7 27.6 7 42.1 0 99.4-112.8 180-252 180s-252-80.6-252-180c0-14.5 2.4-28.6 7-42.1A72.01 72.01 0 0 1 296 404c27.1 0 50.6 14.9 62.9 37 36.2-19.8 80.2-32.8 128.1-36.1l58.4-131.1c4.3-9.8 15.2-14.8 25.5-11.8l91.6 26.5a54.03 54.03 0 0 1 101.6 25.6c0 29.8-24.2 54-54 54-23.5 0-43.5-15.1-50.9-36.1L577 308.3l-43 96.5c49.1 3 94.2 16.1 131.2 36.3 12.3-22.1 35.8-37 62.9-37 39.8 0 72 32.2 72 72-.1 29.3-17.8 54.6-43.1 65.8zm-171.3 83c-14.9 11.7-44.3 24.3-73.7 24.3s-58.9-12.6-73.7-24.3c-9.3-7.3-22.7-5.7-30 3.6-7.3 9.3-5.7 22.7 3.6 30 25.7 20.4 65 33.5 100.1 33.5 35.1 0 74.4-13.1 100.2-33.5 9.3-7.3 10.9-20.8 3.6-30a21.46 21.46 0 0 0-30.1-3.6zM296 440a35.98 35.98 0 0 0-13.4 69.4c11.5-18.1 27.1-34.5 45.9-48.8A35.9 35.9 0 0 0 296 440zm72 108a36 36 0 1 0 72 0 36 36 0 1 0-72 0z"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
